ui-app/client/src/components/PageEditor/functions/EventDefGenerator.ts

```diff
@@ -0,0 +1,222 @@
+import { duplicate } from '@fincity/kirun-js';
+import { PageDefinition } from '../../../types/common';
+import { shortUUID } from '../../../util/shortUUID';
+
+interface buttonDef {
+	key: string;
+	name: string;
+	type: string;
+	properties: {
+		[key: string]: any;
+	};
+	styleProperties: {
+		[key: string]: {
+			resolutions: {
+				ALL: {
+					width: {
+						value: string;
+					};
+				};
+			};
+		};
+	};
+	override: boolean;
+	displayOrder: number;
+}
+
+export const generateFormEvents = (
+	pageDefinition: PageDefinition,
+	componentKey: string | undefined,
+	formName: string,
+	eachFormDef: any,
+	submitButtonKey: string,
+	clearButtonKey: string,
+	selectedFormComponent: Array<string>,
+	buttonName: Array<number>,
```

Comment:
create two different parameters genSubmit, genClear instead of an array. Restructure the code to make it efficient.